tuf/client_rework/updater_rework.py

```diff
@@ -195,318 +215,113 @@ def download_target(
-                )
-
-        if self._metadata.get("snapshot"):
-            if (
-                intermediate_timestamp.snapshot.version
-                <= self._metadata["timestamp"].snapshot["version"]
-            ):
-                raise exceptions.ReplayedMetadataError(
-                    "root",
-                    intermediate_timestamp.snapshot.version(),
-                    self._metadata["snapshot"].version(),
-                )
-
-        intermediate_timestamp.expires()
-
-        return intermediate_timestamp
-
-    def _verify_snapshot(self, file_content: bytes) -> SnapshotWrapper:
-        """
-        TODO
-        """
-
-        # Check against timestamp metadata
-        if self._metadata["timestamp"].snapshot.get("hash"):
-            _check_hashes(
-                file_content, self._metadata["timestamp"].snapshot.get("hash")
-            )
-
-        intermediate_snapshot = SnapshotWrapper.from_json_object(file_content)
-
-        if (
-            intermediate_snapshot.version
-            != self._metadata["timestamp"].snapshot["version"]
-        ):
-            raise exceptions.BadVersionNumberError
-
-        # Check for an arbitrary software attack
-        trusted_root = self._metadata["root"]
-        intermediate_snapshot.verify(
-            trusted_root.keys("snapshot"), trusted_root.threshold("snapshot")
-        )
-
-        # Check for a rollback attack
-        if self._metadata.get("snapshot"):
-            for target_role in intermediate_snapshot.signed.meta:
-                if (
-                    target_role["version"]
-                    != self._metadata["snapshot"].meta[target_role]["version"]
-                ):
-                    raise exceptions.BadVersionNumberError
-
-        intermediate_snapshot.expires()
-
-        return intermediate_snapshot
-
-    def _verify_targets(
-        self, file_content: bytes, filename: str, parent_role: str
-    ) -> TargetsWrapper:
-        """
-        TODO
-        """
-
-        # Check against timestamp metadata
-        if self._metadata["snapshot"].role(filename).get("hash"):
-            _check_hashes(
-                file_content, self._metadata["snapshot"].targets.get("hash")
-            )
-
-        intermediate_targets = TargetsWrapper.from_json_object(file_content)
-        if (
-            intermediate_targets.version
-            != self._metadata["snapshot"].role(filename)["version"]
-        ):
-            raise exceptions.BadVersionNumberError
-
-        # Check for an arbitrary software attack
-        parent_role = self._metadata[parent_role]
-
-        intermediate_targets.verify(
-            parent_role.keys(filename), parent_role.threshold(filename)
-        )
-
-        intermediate_targets.expires()
-
-        return intermediate_targets
+            data = self._load_local_metadata(role)
+            self._bundle.update_delegated_targets(data, role, parent_role)
```

Comment:
Using `self._bundle.update_delegated_targets` here leaves `MetadataBundle.update_targets()` unused.
Not a big deal but mentioning it.